### PR TITLE
Add pointer tests

### PR DESCRIPTION
**Pull request by hainest**

### CODE REVIEW

**Comment by hainest** [comment]
Once we merge #44, I'll add checks for the `pointer_indirections` field.

File: source/parser/x86_64/classifiers.hpp [hunk 1]
```diff
@@ -74,7 +74,6 @@ namespace smeagle::x86_64 {
         }
         // This is NOT correct.
         // TODO It should be struct{T r,i;};, but we don't handle aggregates yet
```
Comment (vsoch) [inline source/parser/x86_64/classifiers.hpp:76]: Is this comment still the case?

File: test/source/libs/gen_allocation.py [hunk 2]
```diff
@@ -121,48 +121,112 @@ def make_funcs(file, category, types):
...
+  SUBCASE("void") {
+    auto const& func = get_one(corpus, "test_ptr_ptr_void");
+    auto const& parameters = func.parameters;
+    CHECK(parameters[0].location == "%rdi");
+    CHECK(parameters[0].type == "Pointer64");
+  }
 }
 """)
+
+# ------------------------------------------ #
```
Comment (vsoch) [inline test/source/libs/gen_allocation.py:223]: Huh, what's the benefit of writing this at runtime vs. just having the file?

Comment (hainest) [reply test/source/libs/gen_allocation.py:223]: I added <10 lines of Python to generate hundreds of lines of C++. When we get reference working, we'll have the same benefit.

File: source/parser/x86_64/classifiers.hpp (hunk 1 above)
Comment (hainest) [reply source/parser/x86_64/classifiers.hpp:76]: Yup. It's something we need to work on. It's a toss-up between that and enums as the next step. I need to get references and constants working in Dyninst before we can add them to Smeagle.

File: test/source/libs/gen_allocation.py (hunk 2 above)
Comment (vsoch) [reply test/source/libs/gen_allocation.py:223]: haha okay, that makes sense! Let me know when you've added the pointer check here and I'll review again, and then can we chat again when you are able about next TODO?

**Comment by hainest** [comment]
I added the checks for the pointer counts. Once this passes the automated tests, I'm good to merge (via squash is recommended). I have some time this afternoon to chat about next items, if you do. Otherwise, I'm free tomorrow except from 1-4 CST.

**Review by vsoch — APPROVED** [review]

**Pull request merged**